includes/wc-payment-api/class-wc-payments-api-client.php

```diff
@@ -277,7 +277,7 @@ public function create_intention(
 		$request['capture_method']       = $capture_method;
 
 		if ( Fraud_Prevention_Service::get_instance()->is_enabled() ) {
-			$request['metadata'] = array_merge( isset( $request['metadata'] ) ? $request['metadata'] : [], $this->get_fingerprint_metadata() );
+			$request['metadata'] = $this->get_fingerprint_metadata();
```

Comment:
According to `psalm` we aren't defining a `metadata` index before this line, so we directly assign `get_fingerprint_metadata` to the `metadata`

Reply:
I think this is ok. If someone will make a change that introduces the metadata at some point then the tests should pick up and indicate the person to merge the metadatas.